File: ProjectManager/Models/Dto/Task.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ProjectManager.Models.Dto
{
    public enum Statuses
    {
        New,
        InProgress,
        Done
    }
    public class Task
    {
        public int TaskId { get; set; }
        public string Name { get; set; }
        public string assignId { get; set; }
        public DateTime DeadLine { get; set; }
        public int ProjectId { get; set; }
        public Statuses Status { get; set; }
        public Project Project { get; set; }

    }
    public class TaskDto : Task
    {
        public string assign { get; set; }

        public TaskDto(Task task)
        {
            this.assignId = task.assignId;
            this.DeadLine = task.DeadLine;
            this.Name = task.Name;
            this.ProjectId = task.ProjectId;
            this.Status = task.Status;
            this.TaskId = task.TaskId;
            this.Project = task.Project;
        }

    }
}

File: ProjectManager/ClientApp/src/app/projects-list/task-list/task.ts

import { User } from "src/app/modules";
import { Project } from "../Project";
export enum Statuses {
    New,
    InProgress,
    Done
  }

export class Task {
    taskId?: Number;
    projectId: number;
    project: Project;
    name: string;
    assign?: User | string;  
    assignId?: String;
    deadLine: Date;
    status: Statuses;

    constructor( ProjectId: number,project:Project, Name: string, DeadLine: Date,status: Statuses, assign?:User,TaskId?:Number) {
        this.taskId = TaskId;
        this.projectId = ProjectId;
        this.name = Name;
        this.assign = assign;      
        this.deadLine = DeadLine;
        this.status = status;
        this.project = project;  
        this.assignId = assign?.id;   
    }
   

}